### DISCUSSION TITLE
How to avoid the duplicate request error when reusing a step in cucumber

### DISCUSSION THREAD

**Discussion by MeganOzanne** [pactumjs/pactum#269] [Q&A] [answered]
Hi, I am using cucumber and Pactumjs to write some API tests. I want to reuse my when step across multiple steps as this is when I make the request to the endpoint I am testing. How can I avoid `Error: Duplicate request initiated.`.

My given steps within my feature file specify the parameters to pass into the request, this is working as expected. My then statements are asserting on what is returned from the request, these are also working as expected. It is just my when statement that is failing.

This is my when statement code:
```js
import { When } from "@cucumber/cucumber";
import { spec, request } from "pactum";
import { search } from "../../support/models/urls";

const sp = spec()

request.setDefaultTimeout(6000);

When("Alice makes a request to the API", async function () {
  sp.post(search.bulkSearch)
    .withBody({
      service: "XYZ",
      conditions: [
        {
          field: "something",
          terms: [`${this.something}`],
          operator: "or",
          use_wildcards: true,
        },
      ],
      countries: [`${this.country}`],
      return_profile_count: 300,
      count: `${this.count}`,
      output_presigned_url: "$S{presignedUrl}",
    });
  this.response = await sp.toss();
});
```

Any help would be greatly appreciated. Thank you in advance.

**Comment by MeganOzanne** [comment]
I have also attempted using the recommended data template approach: 

` stash.loadData('./data');  
  sp.post(search.bulkSearch)
  .withJson({
    '@DATA:TEMPLATE@': 'search:no-country',
    '@OVERRIDES@': {
      terms: [`${this.expertise}`],
      countries: [`${this.country}`],
      count: `${this.count}`,
    }
  });` 

this still returns the duplicate error

**Reply by ASaiAnudeep** [reply]
I don't think data templates and duplicate request error are related.

**Comment by MeganOzanne** [comment]
I have now also attempted following the examples [here](https://github.com/pactumjs/pactum-cucumber-boilerplate) but all tests still return duplicate error.

The only way I can get this to run successfully is by creating new steps for each test but this defeats the point of writing tests in this manner and is not an acceptable solution. Someone must have figured out a way to make the request step reusable. Any help would be greatly appreciated.

**Comment by ASaiAnudeep** [comment, marked as answer]
Hope this discussion would help you answer the question - https://github.com/pactumjs/pactum/discussions/172

**Reply by MeganOzanne** [reply]
Thanks, this was the solution.
